aws/sdk/integration-tests/s3/tests/concurrency.rs

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::model::{Delete, ObjectIdentifier};
+use aws_sdk_s3::types::ByteStream;
+use aws_sdk_s3::Client;
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_http::endpoint::Endpoint;
+use aws_smithy_types::timeout::TimeoutConfig;
+use aws_types::credentials::SharedCredentialsProvider;
+use aws_types::region::Region;
+use aws_types::{Credentials, SdkConfig};
+use bytes::BytesMut;
+use futures::future;
+use hdrhistogram::sync::SyncHistogram;
+use hdrhistogram::Histogram;
+use std::fs::File;
+use std::future::Future;
+use std::iter::repeat_with;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+use tokio::time::{Duration, Instant};
+use tracing::{debug, debug_span, info, Dispatch, Instrument};
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::{filter, EnvFilter, Layer};
+
+#[tokio::test(flavor = "current_thread")]
+async fn test_concurrency_on_multi_thread() {
+    test_concurrency_with_dummy_server().await
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn test_concurrency_on_single_thread() {
+    test_concurrency_with_dummy_server().await
+}
```

Comment:
maybe we should refactor the test to be parameterized on the `http_connector`? Or you could parameterize it on `fn(SdkConfig::Builder) -> SdkConfig::Builder`?